cmdline.rs

```diff
@@ -0,0 +1,69 @@
+//! Command line-configurable transparent proxy.
+//!
+//! Useful for testing e.g. address translation or DNS related functionalities,
+//! such as alterations of hostname mapping combined with IP changes.
+//!
+//! How to use:
+//! cargo run --example cmdline -- [node_addr] [proxy_addr]
+//! This starts a proxy that forwards all traffic between a driver and [proxy_addr] to [node_addr].
+
+use std::{
+    io::Read as _,
+    net::{Ipv4Addr, SocketAddr},
+};
+
+use scylla_proxy::{Node, Proxy, ShardAwareness};
+use tracing::instrument::WithSubscriber;
+
+fn init_logger() {
+    tracing_subscriber::fmt::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .without_time()
+        .init();
+}
+
+fn pause() {
+    println!("Press Enter to stop proxy...");
+    std::io::stdin().read_exact(&mut [0]).unwrap();
+    println!();
+}
+
+#[tokio::main]
+async fn main() {
+    init_logger();
+    let mut args = std::env::args();
+    args.next(); // skip name
+    let real_octet = args.next();
+    let proxy_octet = args.next();
+    let node_real_addr = SocketAddr::new(
+        std::net::IpAddr::V4(Ipv4Addr::new(
+            172,
+            42,
+            0,
+            real_octet.as_ref().map(|s| s.parse().unwrap()).unwrap_or(2),
+        )),
+        9042,
+    );
+    let node_proxy_addr = SocketAddr::new(
+        std::net::IpAddr::V4(Ipv4Addr::new(
+            127,
+            0,
+            0,
+            proxy_octet.as_ref().map(|s| s.parse().unwrap()).unwrap_or(66),
+        )),
+        9042,
+    );
```

Comment:
I think it is a bit confusing that the `[node_addr]` and `[proxy_addr]` arguments are actually only the last octet of the IP address. I think we should try to parse `[node_addr]` and `[proxy_addr]` as full addresses.